### PR TITLE
SYCL CI: Avoid setvars.sh

### PR DESCRIPTION
**Pull request by masterleinad**
This PR tries to avoid the SYCL CI issues we are seeing when running `setvars.sh` by setting the necessary environment variables in `Dockerfile` already.

### CODE REVIEW

**Comment by masterleinad** [comment]
Seems to work.

File: scripts/docker/Dockerfile.sycl [hunk 1]
```diff
@@ -55,3 +55,12 @@ RUN wget https://registrationcenter-download.intel.com/akdlm/irc_nas/19133/l_one
...
     rm l_oneDPL_p_2022.0.0.25335.sh
+
+# clang++
+ENV PATH=/opt/intel/oneapi/compiler/latest/linux/bin-llvm/:$PATH
+# sycl-ls, icpx
+ENV PATH=/opt/intel/oneapi/compiler/latest/linux/bin/:$PATH
+# libsycl
+ENV LD_LIBRARY_PATH=/opt/intel/oneapi/compiler/latest/linux/lib:$LD_LIBRARY_PATH
+# libsvml
+ENV LD_LIBRARY_PATH=/opt/intel/oneapi/compiler/latest/linux/compiler/lib/intel64_lin:$LD_LIBRARY_PATH
```
Comment (dalg24) [inline scripts/docker/Dockerfile.sycl:66]: Did you consider doing something like
```
RUN echo 'source /opt/intel/oneapi/setvars.sh --force' > $HOME/.bashrc
```
instead.
Is there still no "official" container we could use?

Comment (masterleinad) [reply scripts/docker/Dockerfile.sycl:66]: This is the same approach as in https://github.com/kokkos/ci-containers/blob/main/ubuntu which has been working well and we do the same for, say `CMake` in `Dockerfiles` already.
Last I checked, we would not parse any of the `.bashrc`, `.profile`, etc. files when testing with `jenkins`. An alternative might be messing with ENTRYPOINT but that is way more messy in my opinion.

I'm not aware of an "official" container that would contain the Codeplay plugin.

**Review by dalg24 — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Comment by crtrott** [comment]
SYCL passed in jenkins

**Pull request merged**